file/src/main/scala/akka/stream/alpakka/file/impl/archive/TarReaderStage.scala

```diff
@@ -68,18 +68,26 @@ private[file] class TarReaderStage
           } else new CollectFile(metadata, buffer)
         }
 
-        val metadata = TarArchiveEntry.parse(headerBuffer)
-        val buffer = headerBuffer.drop(TarArchiveEntry.headerLength)
-        if (isAvailable(flowOut)) {
-          pushSource(metadata, buffer)
+        if (headerBuffer.head == 0) {
+          log.debug("empty filename, detected EOF padding, completing")
```

Comment:
Hard to find any information about that. I guess lots of historic reasons and tape stuff...